pallets/treasury/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+
+use polymesh_primitives::{AccountKey, Beneficiary, IdentityId};
+use polymesh_runtime_balances as balances;
+use polymesh_runtime_common::{
+    traits::{
+        balances::Trait as BalancesTrait, identity::Trait as IdentityTrait, CommonTrait,
+        NegativeImbalance,
+    },
+    Context,
+};
+use polymesh_runtime_identity as identity;
+
+use codec::Encode;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage,
+    dispatch::DispatchResult,
+    ensure,
+    traits::{Currency, ExistenceRequirement, Imbalance, OnUnbalanced, WithdrawReason},
+};
+use frame_system::{self as system, ensure_root, ensure_signed};
+use sp_runtime::traits::Saturating;
+use sp_std::{convert::TryFrom, prelude::*};
+
+pub type ProposalIndex = u32;
+
+type BalanceOf<T> = <T as CommonTrait>::Balance;
+type Identity<T> = identity::Module<T>;
+
+pub trait Trait: frame_system::Trait + CommonTrait + BalancesTrait + IdentityTrait {
+    // The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
+}
+
+pub trait TreasuryTrait<Balance> {
+    fn disbursement(target: IdentityId, amount: Balance);
+    fn balance() -> Balance;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Treasury {
+        pub Balance get(fn balance) config(): BalanceOf<T>;
```

Comment:
why we drop the approach of having an account for the treasury?

Reply:
1. There is no initialization dependency between modules. My first approach was that, and I got some weird behaviour between Identity module, balance module and the new one.
2. Each fee transfer does not emit events for regular transfer between account. Actually during my first testing phase, I had to duplicate a lot of code from balances module to avoid the emission of those events.